### PR TITLE
spidermonkey: drop from all-packages, add to aliases

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

See https://github.com/NixOS/nixpkgs/pull/100026#issuecomment-705660430

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by worldofpeace** [comment]
Thanks for looking out @mkg20001

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -661,6 +661,8 @@ mapAliases ({
   youtubeDL = youtube-dl;  # added 2014-10-26
   zdfmediathk = mediathekview; # added 2019-01-19
   gnome_user_docs = gnome-user-docs; # added 2019-11-20
+  # spidermonkey is not ABI upwards-ompatible, so only allow this for nix-shell
+  spidermonkey = spidermonkey_78; # added 2020-10-09
```
Comment (grahamc) [inline pkgs/top-level/aliases.nix:665]: A brief note that aliases.nix are enabled by default for users of nixpkgs, and their use is only prohibited from within nixpkgs itself. In other words, `nix-shell` isn't really the determining factor here.

**Comment by doronbehar** [comment]
@mkg20001 @worldofpeace This PR causes eval errors for all PRs now, due to ofborg's "no aliases check". It's the 2nd ofborg eval causing commit this day! (https://github.com/NixOS/nixpkgs/commit/cd1b59476793278191c37572c8081b856512eec4)

To do this change properly, you should iterate over all usages of spidermonkey and remove any usage of the alias.

**Comment by worldofpeace** [comment]
Yeah, ofborg has been... Anyways, we can fix it by just replacing uses of `spidermonkey` in nixpkgs with the versioned attribute.

**Comment by doronbehar** [comment]
Fixed in 0da82de388a5214522262a3885080bad1c02fd1b .

**Pull request merged**